controllers/metrics.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2019 microsoft.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import(
+	"time"
+	"github.com/prometheus/client_golang/prometheus"
+	models "github.com/xinsnake/databricks-sdk-golang/azure/models"
+)
+
+func trackJobExecutionTime(histogram prometheus.Histogram, f func() (models.Job, error)) (models.Job, error) {
+	startTime := time.Now()
+
+	defer trackMetric(startTime, histogram)
+
+	return f()
+}
+
+func trackExecutionTime(histogram prometheus.Histogram, f func() error) error {
+	startTime := time.Now()
+
+	defer trackMetric(startTime, histogram)
+
+	return f()
+}
+
+func trackMetric(startTime time.Time, histogram prometheus.Histogram) {
+	endTime := float64(time.Now().Sub(startTime).Nanoseconds() / int64(time.Millisecond))
```

Comment:
nit: The `sub` method on `time` produces a `duration` which has the method [Milliseconds](https://godoc.org/time#Duration.Milliseconds). I'd favor using that over returning `nanosec`.

Reply:
ah, the reason was because `Sub().Milliseconds` gave me an int64 rather than a float64. I've reworked it now

Reply:
Unfortunately not, there are no milliseconds and this change has been rejected multiple times: https://github.com/golang/go/issues/28564

Reply:
Finally got to the bottom of this, yes there is a `Milliseconds` method, but only in golang v1.13 and this project currently targets 1.12 in its dev container. https://golang.org/doc/go1.13#time

Reply:
nit: In `time.Now().Sub(startTime).Nanoseconds()` I think you can replace it with `time.Now().Sub(startTime)`. 
nit: I don't think the int64 conversion is needed if you don't call Nanoseconds
nit: the variable is called `endTime` but looks more like `duration`?
Also, is the float conversion designed to allow sub-millisecond resolution in the metrics? If so, I'm not sure that the conversion on the outside will achieve that.

```suggestion
duration := float64(time.Now().Sub(startTime) / time.Millisecond)
```